src/main/java/com/teamwizardry/librarianlib/features/animator/Animation.kt

```diff
@@ -139,5 +150,42 @@ abstract class Animation<T : Any>(val target: T, val property: AnimatablePropert
 
     internal var _id: Int = -1
         private set
+
+    // builder methods
+
+    fun completion(fn: Runnable): Animation<T> {
+        this.completion = fn
+        return this
+    }
+
+    fun start(time: Float): Animation<T> {
+        this.start = time
+        return this
+    }
+
+    fun duration(time: Float): Animation<T> {
+        this.duration = time
+        return this
+    }
+
+    fun repeat(count: Int): Animation<T> {
+        this.repeatCount = count
+        return this
+    }
+
+    fun reverseOnRepeat(): Animation<T> {
+        this.shouldReverse = true
+        return this
+    }
+
+    fun useAbsoluteTime(): Animation<T> {
+        this.useAbsoluteTime = true
+        return this
+    }
+
+    fun addTo(animator: Animator): Animation<T> {
+        animator.add(this)
+        return this
+    }
```

Comment:
```suggestion
    }
    //endregion
```
(with the other suggestion to open the region of course)